mix/exit_layer.nim

```diff
@@ -0,0 +1,104 @@
+import std/[enumerate, strutils]
+import chronicles, chronos, metrics
+import libp2p, libp2p/[builders, stream/connection]
+import ./[mix_metrics, exit_connection, serialization, utils]
+
+type ProtocolHandler* = proc(conn: Connection, codec: string): Future[void] {.
+  async: (raises: [CancelledError])
+.}
+
+type ExitLayer* = object
```

Comment:
Suggest putting in doc comments.

Reply:
I'll try to add comments on all code once refactoring is complete. I'm still not sure if this will be its final form